pkg/apis/sparkoperator.k8s.io/v1beta2/defaults.go

```diff
@@ -70,6 +70,6 @@ func setExecutorSpecDefaults(spec *ExecutorSpec, sparkConf map[string]string) {
 	}
 	if _, exists := sparkConf["spark.executor.instances"]; !exists && spec.Instances == nil {
 		spec.Instances = new(int32)
-		*spec.Instances = 1
+		*spec.Instances = 0
```

Comment:
Does setting the `initialExecutors: 0` in the spark application yaml make this unncessary? Or vice versa, set it here and not the other place. Do we need both?

Reply:
It's possible to use either or, and we don't force users into one or the other, they're free to modify their values file as they see fit. But in the case of static allocation without a defined default, we don't want to default to spinning up an unnecessary executor

Reply:
I feel like defaulting to 1 is totally reasonable if dynamic allocation ISNT set.  I'd be more inclined to modify the condition to not set to 1 when we're using dynamic allocation

Reply:
Updated